Formula/gomuks.rb

```diff
@@ -0,0 +1,18 @@
+class Gomuks < Formula
+  desc "Terminal based Matrix client written in Go"
+  homepage "https://maunium.net/go/gomuks"
+  url "https://github.com/tulir/gomuks/archive/v0.1.1.tar.gz"
+  sha256 "e1fd00a06a6145f72e134c9e1cddab53828716914a8ea61047f77aab16bc9679"
+
+  depends_on "go" => [:build, "1.13"]
+  depends_on "libolm"
+
+  def install
+    system "go", "build", *std_go_args, "-ldflags", "-s -w"
+    doc.install "README.md"
+  end
+
+  test do
+    assert_equal "gomuks version v#{version}", shell_output("#{bin}/gomuks --version").strip
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
Yeah I understand but the problem is it is a TUI type program. There is no other functions that I can run. Attempting to run the program itself in the test causes it to crash.

```
panic: terminal not cursor addressable [recovered]
	panic: terminal not cursor addressable

goroutine 1 [running]:
maunium.net/go/gomuks/debug.Recover()
	/private/tmp/gomuks-20200611-99679-ogbklp/gomuks-0.1.0/debug/debug.go:109 +0x94
panic(0x4782520, 0xc00017b870)
	/usr/local/Cellar/go/1.14.3/libexec/src/runtime/panic.go:969 +0x166
main.(*Gomuks).Start(0xc0005b91a0)
	/private/tmp/gomuks-20200611-99679-ogbklp/gomuks-0.1.0/gomuks.go:116 +0x149
main.main()
	/private/tmp/gomuks-20200611-99679-ogbklp/gomuks-0.1.0/main.go:77 +0x484
```

Reply:
That should probably be reported upstream so they can just tell people this without a crash. After that you can check for that message